src/js/view/popup/popupWrite.js

```diff
@@ -0,0 +1,372 @@
+    var bounds = util.map(guideElements, function(element) {
+        return element.getBoundingClientRect();
+    });
+    var boundsInFirstRow = util.filter(bounds, function(bound) {
+        return bounds[0].top === bound.top;
+    });
+    var top = Number.MAX_VALUE;
+    var left = Number.MAX_VALUE;
+    var bottom = 0;
+    var right = 0;
+    var x, y, arrowDirection, arrowPoint;
+
+    util.forEach(boundsInFirstRow, function(bound) {
+        top = mmin(top, bound.top);
+        left = mmin(left, bound.left);
+        bottom = mmax(bottom, bound.bottom);
+        right = mmax(right, bound.right);
+    });
+
+    if (top >= layerHeight) {
+        y = top - layerHeight - 3;
+        arrowDirection = 'bottom';
+    } else {
+        y = bottom + 4;
+        arrowDirection = 'top';
+    }
+
+    x = (left + right - layerWidth) / 2;
+
+    if (x + layerWidth >= window.innerWidth) {
+        arrowPoint = (right - x + 2);
+        x = window.innerWidth - layerWidth;
+    }
+
+    return {
+        x: x - parentRect.left,
+        y: y - parentRect.top,
+        arrow: {
+            direction: arrowDirection,
+            point: arrowPoint
+        }
+    };
+};
+
+PopupWrite.prototype._setArrowDirection = function(arrow) {
+    var borderElement = document.getElementById('arrow-border');
+    var fillElement = document.getElementById('arrow-fill');
+
+    if ((arrow.direction === 'top') && (domutil.hasClass(borderElement, config.classname('popup-bottom')))) {
+        domutil.removeClass(borderElement, config.classname('popup-bottom'));
+        domutil.removeClass(fillElement, config.classname('popup-bottom'));
+        domutil.addClass(borderElement, config.classname('popup-top'));
+        domutil.addClass(fillElement, config.classname('popup-top'));
+    } else if ((arrow.direction === 'bottom') && (domutil.hasClass(borderElement, config.classname('popup-top')))) {
+        domutil.removeClass(borderElement, config.classname('popup-top'));
+        domutil.removeClass(fillElement, config.classname('popup-top'));
+        domutil.addClass(borderElement, config.classname('popup-bottom'));
+        domutil.addClass(fillElement, config.classname('popup-bottom'));
+    }
+
+    if (arrow.point) {
+        borderElement.style.left = arrow.point + 'px';
+    }
+};
+
+PopupWrite.prototype._createDatepicker = function(start, end) {
+    this.rangePicker = DatePicker.createRangePicker({
+        startpicker: {
+            date: new Date(start),
+            input: '#startpicker-input',
+            container: '#startpicker-container'
+        },
+        endpicker: {
+            date: new Date(end),
+            input: '#endpicker-input',
+            container: '#endpicker-container'
+        },
+        format: 'yyyy-MM-dd HH:mm',
+        timepicker: {
+            showMeridiem: false
+        },
+        usageStatistics: false
+    });
+    this.rangePicker.getStartpicker()._$container[0].style.zIndex = this.layer.zIndex + 2;
```

Comment:
Is there any way to use official API? Why don't you ask `tui-date-picker` for this?